crates/re_entity_db/src/entity_db.rs

```diff
@@ -83,14 +83,31 @@ fn insert_row_with_retries(
 /// NOTE: all mutation is to be done via public functions!
 pub struct EntityDb {
     /// Set by whomever created this [`EntityDb`].
+    ///
+    /// Clones of an [`EntityDb`] gets a `None` source.
     pub data_source: Option<re_smart_channel::SmartChannelSource>,
 
     /// Comes in a special message, [`LogMsg::SetStoreInfo`].
     set_store_info: Option<SetStoreInfo>,
 
+    /// If this entity db is the result of a clone, which store was it cloned from?
+    ///
+    /// A cloned store always gets a new unique ID.
+    ///
+    /// We currently only use entity db cloning for blueprints:
+    /// when we receive a _default_ blueprints on the wire (e.g. from a recording),
+    /// we clone it and make the clone the _active_ blueprint.
+    /// This means all active blueprints are clones.
+    cloned_from: Option<StoreId>,
```

Comment:
I think that would be smart… except `SetStoreInfo` can be `None` :/

Reply:
I would actually love to make it never None, though it's a bigger refactor that involves occasionally queuing incoming messages in a "pre-store" queue until we've received the SetStoreInfo message.

In practice SetStoreInfo **should** always be the first message in the stream. The only reason it can end up as `None` is if something went very wrong. I would actually categorize this as some form of comms/protocol error and say that a Store missing a StoreInfo is much more likely to be incomplete and/or corrupt than it is to be useful.  This holds even moreso for Blueprints.

Reply:
I moved `cloned_from` to `StoreInfo`